app/uk/gov/hmrc/agentservicesaccount/forms/UpdateMoneyLaunderingSupervisionForm.scala

```diff
@@ -30,39 +30,103 @@ object UpdateMoneyLaunderingSupervisionForm {
   private val supervisoryNumberRegex = """^[A-Za-z0-9\,\.\'\-\/\ ]{0,200}$""".r // remove all spaces from input before matching to ensure correct digit count
   private val trimmedText = text.transform[String](x => x.trim, x => x)
 
-  private val invalidDateConstraint: Constraint[(String, String, String)] = Constraint[(String, String, String)] { data: (String, String, String) =>
-    val (day, month, year)  = data
-
+  private def invalidDateCheck(day: String, month: String, year: String): Either[(String, String), String] = {
     Try {
       require(year.length == 4, "Year must be 4 digits")
       LocalDate.of(year.toInt, month.toInt, day.toInt)
     } match {
-      case Failure(_) => Invalid(ValidationError("update-money-laundering-supervisory.error.date.invalid"))
-      case Success(_) => Valid
+      case Failure(_) => Left("day" -> "update-money-laundering-supervisory.error.date.invalid")
+      case Success(_) => Right("")
```

Comment:
Is there a better type we could use rather than an empty string?

Reply:
Yes, but it was late. Alternatives are welcome. I had thought of using a `Boolean` (after closing the laptop) but it's something that is discarded in the calling code.

Reply:
Agreed, that makes more sense.